File: FrontEnd/src/app/models/requests/search-words.request.ts
import { SearchWordsOrderBy } from '../enums/search-words-order-by.enum';
import { SortDirection } from '../enums/sort-direction.enum';

export interface SearchWordsRequest {
  offset: number;
  limit: number;
  filter: string;
  orderBy: SearchWordsOrderBy;
  direction: SortDirection;
}

File: BackEnd/Api/Endpoints/Words/Search/SearchWordsRequest.cs
using OhMyWord.Core.Services;

namespace OhMyWord.Api.Endpoints.Words.Search;

public class SearchWordsRequest
{
    public int Offset { get; init; } = IWordsRepository.OffsetMinimum;
    public int Limit { get; init; } = IWordsRepository.LimitDefault;
    public string Filter { get; init; } = string.Empty;
    public string OrderBy { get; init; } = string.Empty;
    public bool IsDescending { get; init; }
}
